tests/routers/test_score_set.py

```diff
@@ -1256,6 +1256,119 @@ def test_multiple_score_set_meta_analysis_multiple_experiment_sets_different_sco
+    jsonschema.validate(instance=score_set_2, schema=ScoreSetCreate.schema())
+
+    response = client.post("/api/v1/score-sets/", json=score_set_2)
+    response_data = response.json()
+    assert response.status_code == 403
+    assert "Score sets may not be added to a meta-analysis experiment." in response_data["detail"]
+
+
+def test_create_single_score_set_meta_analysis_to_others_score_set(session, data_provider, client, setup_router_db, data_files):
+    experiment = create_experiment(client)
+    score_set = create_seq_score_set_with_variants(
+        client, session, data_provider, experiment["urn"], data_files / "scores.csv"
+    )
+
+    score_set = (client.post(f"/api/v1/score-sets/{score_set['urn']}/publish")).json()
+    change_ownership(session, score_set["urn"], ScoreSetDbModel)
+    meta_score_set = create_seq_score_set_with_variants(
+        client,
+        session,
+        data_provider,
+        None,
+        data_files / "scores.csv",
+        update={"title": "Test Meta Analysis", "metaAnalyzesScoreSetUrns": [score_set["urn"]]},
+    )
+
+    score_set_refresh = (client.get(f"/api/v1/score-sets/{score_set['urn']}")).json()
+    assert meta_score_set["metaAnalyzesScoreSetUrns"] == [score_set["urn"]]
+    assert score_set_refresh["metaAnalyzedByScoreSetUrns"] == [meta_score_set["urn"]]
+    assert isinstance(MAVEDB_TMP_URN_RE.fullmatch(meta_score_set["urn"]), re.Match)
+
+
+def test_multiple_score_set_meta_analysis_single_experiment_with_different_creator(
+    session, data_provider, client, setup_router_db, data_files
+):
+    experiment = create_experiment(client)
+    score_set_1 = create_seq_score_set_with_variants(
+        client, session, data_provider, experiment["urn"], data_files / "scores.csv", update={"title": "Score Set 1"}
+    )
+    score_set_2 = create_seq_score_set_with_variants(
+        client, session, data_provider, experiment["urn"], data_files / "scores.csv", update={"title": "Score Set 2"}
+    )
+
+    score_set_1 = (client.post(f"/api/v1/score-sets/{score_set_1['urn']}/publish")).json()
+    score_set_2 = (client.post(f"/api/v1/score-sets/{score_set_2['urn']}/publish")).json()
+
+    change_ownership(session, score_set_2["urn"], ScoreSetDbModel)
+    meta_score_set = create_seq_score_set_with_variants(
+        client,
+        session,
+        data_provider,
+        None,
+        data_files / "scores.csv",
+        update={"title": "Test Meta Analysis", "metaAnalyzesScoreSetUrns": [score_set_1["urn"], score_set_2["urn"]]},
+    )
+    score_set_1_refresh = (client.get(f"/api/v1/score-sets/{score_set_1['urn']}")).json()
+    assert meta_score_set["metaAnalyzesScoreSetUrns"] == sorted([score_set_1["urn"], score_set_2["urn"]])
+    assert score_set_1_refresh["metaAnalyzedByScoreSetUrns"] == [meta_score_set["urn"]]
+
+    meta_score_set = (client.post(f"/api/v1/score-sets/{meta_score_set['urn']}/publish")).json()
+    assert meta_score_set["urn"] == "urn:mavedb:00000001-0-1"
```

Comment:
In so far as we check the URNs of test objects, I think we only need to check that the URN matches the URN RegEx.

Reply:
I added the RegEx checks to all meta-analysis tests, but I support keeping them cause the meta-analysis urn is more special. This way is more accurate. RegEx only confirms they match the basic rules, but can't make sure the middle one is number 0 or letters.